File: python/src/main/python/stores/IceCreamStore.py
from Cone import *
from icecreams.IceCream import *

class IceCreamStore():

    def createIceCream(self, flavor):
        pass

    def prepareCone(self, flavor):
        iceCream = self.createIceCream(flavor)
        cone = Cone()
        cone.prepare()
        cone.addIceCream(iceCream)
        cone.serve()
        return cone

File: java/src/main/java/mx/iteso/stores/IceCreamStore.java
package mx.iteso.stores;

import mx.iteso.Cone;
import mx.iteso.icecreams.IceCream;

/**
 * Abstract class IceCreamStore.
 */
public abstract class IceCreamStore {
    /**
     * Create Ice Cream.
     *
     * @param flavor flavor.
     * @return IceCream
     */
    public abstract IceCream createIceCream(String flavor);

    /**
     * Prepare Cone.
     *
     * @param flavor ice cream flavor
     * @return Cone
     */
    public Cone prepareCone(final String flavor) {
        IceCream iceCream = createIceCream(flavor);
        Cone cone = new Cone();
        cone.prepare();
        cone.addIceCream(iceCream);
        cone.serve();
        return cone;
    }
}
